### ISSUE TITLE
Model doesn't get saved during training

### ISSUE THREAD

**Issue by Iris-and-the-machines** [janclemenslab/das#71]
Hi Jan :)
after a year of not using das I know experience a problem I can't fix:
When training a model from the gui, the model doesn't get saved. I get the following error message after each epoch:
20/20 - 4s - loss: 1.4199
WARNING:tensorflow:Can save best model only with val_loss available, skipping.
WARNING:tensorflow:Can save best model only with val_loss available, skipping.
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss

It seems to be related to the output from keras?! Here someone also had this problem:
[(https://github.com/tensorflow/tensorflow/issues/44107#issuecomment-747389632)]

my installation is:
INFO:das.cli:  Windows-10-10.0.19045-SP0
INFO:das.cli:  DAS v0.32.2
INFO:das.cli:     GUI is available.
INFO:das.cli:     xarray-behave v0.35.2
INFO:das.cli:     pyqtgraph v0.12.4
INFO:das.cli:     PySide2 v5.13.2
INFO:das.cli:     Qt v5.12.9
INFO:das.cli:
INFO:das.cli:  tensorflow v2.6.0
INFO:das.cli:  keras v2.6.0
INFO:das.cli:     GPU is not available.
INFO:das.cli:
INFO:das.cli:  python v3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:39:05) [MSC v.1929 64 bit (AMD64)]
INFO:das.cli:  pandas v2.2.2
INFO:das.cli:  numpy v1.23.5
INFO:das.cli:  h5py v3.9.0
INFO:das.cli:  scipy v1.13.0
INFO:das.cli:  scikit-learn v1.4.2
INFO:das.cli:  xarray v2024.3.0

Do you have any tips on how to solve this?

Thanks :)
Iris

**Comment by postpop** [comment]
Hi Iris :)

Sorry that the models don't get saved!!
Unfortunately, I can't reproduce this with my test data set here - tried it on windows 10, same packages as you have. 

Maybe sth with about data?
During training, you should get outputs that also list the val_loss, like so:
```6/6 - 4s - loss: 0.0208 - val_loss: 0.0336```
In your example, `20/20 - 4s - loss: 1.4199`, there is only the loss but not val_loss.
But I guess your data set does have a validation set, right? In the ".npy" folder produced with "Make dataset for training" should be subfolders called "train", "validation", and potentially also "test". 
If you want to, I can also have a look at the dataset you use.

Have a nice day!!
Jan

**Comment by Iris-and-the-machines** [comment]
Hi Jan,
thanks for the super fast reply! Checking my data sets again now I came to realize that I had set the slit for validation to 0. So I did not have a validation set. Sorry for wasting your time!
And a nice day to you too :)
